### DISCUSSION TITLE
[Knative] Improve Knative Eventing End-to-End Observability by addressing top issues identified by community

### DISCUSSION THREAD

**Discussion by nate-double-u** [cncf/mentoring#618] [Google Summer of Code 2022]
Today, Knative Eventing has some support for observability [1](https://knative.dev/docs/eventing/observability/logging/collecting-logs/) [2](https://knative.dev/docs/eventing/accessing-traces/) [3](https://knative.dev/docs/eventing/troubleshooting/) but it is piecewise and user needs for end-to-end observability are not fully addressed https://github.com/knative/eventing/issues/3387 https://github.com/knative-sandbox/eventing-kafka/issues/29 [6](https://github.com/lionelvillard/kn-trace). The idea is to find out what are the biggest gaps in Knative Eventing by asking the community. Then to find a few issues that are “low hanging” fruits that can be solved during summer with the overall goal to to simplify end-to-end observability by improving support for OpenTelemetry in Knative Eventing and/or create a new plugin(s) for kn CLI. We see that work to be accomplished in few stages. Initial Stage (few weeks) of getting used to knative-eventing with few sample applications, reach out to Knative Eventing community via mailing list and Knative Slack to ask for interviews or feedback on what are biggest problems with observability in Knative Eventing. Then first stage (few weeks) to identify the highest priority problem that can be solved in 1-2 weeks (small size), share the solution, gather feedback. Possible second stage (if time allows) to improve the features from previous stage and/or address larger problem in 3-4 weeks. And final Stage (last weeks of summer) to write Knative docs, blog post, advertise

Mentor(s): Aleksander Slominski @aslom, Ansu Varghese @aavarghese, and Lionel Villard @lionelvillard

https://github.com/cncf/mentoring/blob/main/summerofcode/2022.md#improve-knative-eventing-end-to-end-observability-by-addressing-top-issues-identified-by-community

**Comment by gabo1208** [comment]
I'm interested in this topic to improve my knowledge on observability, One question : I'm already collaborating in other areas in Knative, but I want to learn observability, is ok for me to be a mentee here? cc @nate-double-u

**Reply by aslom** [reply]
I do not know requirements but I expect as long as you are student meeting requirements than you can apply? @nate-double-u  ?

**Reply by nate-double-u** [reply]
Hi @gabo1208, I think that the [GSoC 2022 FAQ](https://developers.google.com/open-source/gsoc/faq) has a question that is similar yours:

[Can someone already participating in open source be a GSoC Contributor?](https://developers.google.com/open-source/gsoc/faq#can_someone_already_participating_in_open_source_be_a_gsoc_contributor)

> The goal of GSoC is to bring new contributors into open source organizations. GSoC can also help beginner contributors learn the ins and outs of open source while being mentored by experienced community members.

> GSoC is for new and beginner contributors to open source, it is not for experienced contributors to open source.

> As an exception, GSoC Contributors who have been accepted once (either in 2020 or 2021) may apply to participate in 2022. This can be with the same organization they participated with before or with a different organization. Applicants should be sure to note their previous relationship with the organization in their proposal.

**Reply by nate-double-u** [reply]
There is more info in the [Program Rules](https://summerofcode.withgoogle.com/rules) regarding eligibility criteria.

**Comment by aslom** [comment]
I thought I would provide a bit of background about why we think this project is worth doing: why should you consider doing this project about Observability? When you look at this project you may ask why is improving observability for Knative Eventing exciting? 

We see that cloud applications are getting more complex, contain many parts, and some may be interacting between serverless functions that are short running and large numbers are created when processing events. Then to be able to understand them becomes very important and that is why great observability is critical for great cloud applications. As it was noticed few months ago “Observability Is the New Kubernetes” and this blog post contains list of current challenges such as dealing with too much of data: https://thenewstack.io/observability-is-the-new-kubernetes/ 

Working on this project you get a deep understanding of modern cloud native event driven and make real improvements for many users of Knative.  

Feel free to talk with us in one of slack channels: #general channel in knative.slack.com or #knative in kubernetes.slack.com or contacting as directly https://twitter.com/aslom
